extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/model/Range.java

```diff
@@ -52,15 +57,15 @@ public Calendar getEnd() {
     /**
      * Returns the duration in minutes.
      */
-    public long getDuration() {
+    public QuantityType<Time> getDuration() {
         if (start == null || end == null) {
-            return -1;
+            return null;
         }
         if (start.after(end)) {
-            return 0;
+            return new QuantityType<Time>(0, SmartHomeUnits.MINUTE);
         }
         long diff = end.getTimeInMillis() - start.getTimeInMillis();
-        return diff / 60000;
+        return new QuantityType<Time>(diff / 60000, SmartHomeUnits.MINUTE);
```

Comment:
You could get rid of the magic number 60000 by using the QuantityType itself for doing the ms to minutes conversion:

```java
long diff = end.getTimeInMillis() - start.getTimeInMillis();
QuantityType millis = new QuantityType<>(diff, MetricPrefix.MILLI(SmartHomeUnits.SECOND));
return millis.toUnit(SmartHomeUnits.MINUTE);
```